clippy_lints/src/collapsible_if.rs

```diff
@@ -285,3 +300,39 @@ fn span_extract_keyword(sm: &SourceMap, span: Span, keyword: &str) -> Option<Spa
         })
         .next()
 }
+
+/// Peel the parentheses of off an `if` expression, e.g. `((if true {} else {}))`.
+fn peel_parens(sm: &SourceMap, mut span: Span) -> (Span, Span, Span) {
+    use crate::rustc_span::Pos;
+    use rustc_span::SpanData;
+
+    let start = span.shrink_to_lo();
+    let end = span.shrink_to_hi();
+
+    loop {
```

Comment:
We could simplify this loop: first we create a function that returns `Option<Span>`, trimming the outermost parens, then this can be a `while let` loop, updating the span.

Reply:
The tricky thing there is that we need to update 3 variables, so it gets kind of verbose. Anyway, I've now solved it using recursion, lmk what you think.